packages/components/icon/src/icon.js

```diff
@@ -0,0 +1,78 @@
+import { TSElement, unsafeCSS, html, customElementDefineHelper } from '@tradeshift/elements';
+import styles from './icon.css';
+
+import { fileIcon, removeIcon, downloadIcon, infoIcon, arrowUpIcon } from './assets/icons';
+
+customElementDefineHelper(
+	'ts-icon',
+	class extends TSElement {
+		constructor() {
+			super();
+			this.size = 'default';
+			this.color = 'gray';
+			this.icon = 'file';
+			this.icons = {
+				fileIcon,
+				removeIcon,
+				downloadIcon,
+				infoIcon,
+				arrowUpIcon
+			};
+			this.statusColors = {
+				default: 'slate',
+				danger: 'red',
+				error: 'red',
+				success: 'green',
+				active: 'blue',
+				focus: 'blue',
+				primary: 'blue',
+				disabled: 'gray-lighter'
+			};
+		}
+
+		static get styles() {
+			return [TSElement.styles, unsafeCSS(styles)];
+		}
+
+		static get properties() {
+			return {
+				icon: { type: String, reflect: true },
+				color: { type: String, reflect: true },
+				colors: { type: Array, reflect: true },
+				size: { type: String, reflect: true },
+				status: { type: String, reflect: true }
+			};
+		}
+
+		get colorA() {
+			if (this.status && this.statusColors[this.status]) {
+				return this.statusColors[this.status];
+			} else {
+				return this.colors ? this.colors[0] : this.color;
+			}
+		}
+
+		get colorB() {
+			//	TODO: add secondary color for different status
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=javascript%3AS1135)